### PR TITLE
Qa v0.2.19

### PR DESCRIPTION
**Pull request by bennsimon**
- Make number of contact schedules shown on summary configurable
- Add `contact_no` to globals for quick check, when its opened from register Activity
- Fix null GA
- Fix incorrect count of profile contact container relevant required fields

### CODE REVIEW

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/repository/PatientRepository.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/util/AncMetadata.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/activity/ContactSummarySendActivity.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5311257&bid=17380449)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/repository/PatientRepository.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/util/AncMetadata.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/activity/ContactSummarySendActivity.java  3
- opensrp-anc/src/main/java/org/smartregister/anc/library/task/LoadContactSummaryDataTask.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5311257&bid=17380449)

File: gradle.properties [hunk 1]
```diff
@@ -11,7 +11,7 @@
 android.debug.obsoleteApi=true
 
 ## PUBLISHING VARS
-VERSION_NAME=2.0.0-SNAPSHOT
+VERSION_NAME=2.0.2.5-LOCAL-SNAPSHOT
```
Comment (dubdabasoduba) [inline gradle.properties:14]: Update this to the correct versioning.

File: opensrp-anc/src/main/java/org/smartregister/anc/library/activity/ContactJsonFormActivity.java [hunk 2]
```diff
@@ -141,6 +142,14 @@ protected void checkBoxWriteValue(String stepName, String parentKey, String chil
...
     @Override
     public void onBackPressed() {
+        if (getmJSONObject().optString(JsonFormConstants.ENCOUNTER_TYPE).equals(ConstantsUtils.JsonFormUtils.ANC_PROFILE_ENCOUNTER_TYPE)) {
+            ContactWizardJsonFormFragment contactWizardJsonFormFragment = (ContactWizardJsonFormFragment) getVisibleFragment();
+            contactWizardJsonFormFragment.getPresenter().validateAndWriteValues();
+            Intent intent = new Intent();
+            intent.putExtra("formInvalidFields",
+                    getmJSONObject().optString(JsonFormConstants.ENCOUNTER_TYPE) + ":" + contactWizardJsonFormFragment.getPresenter().getInvalidFields().size());
+            setResult(RESULT_OK, intent);
+        }
```
Comment (dubdabasoduba) [inline opensrp-anc/src/main/java/org/smartregister/anc/library/activity/ContactJsonFormActivity.java:152]: What is this for? The removal of the required fields count?

File: opensrp-anc/src/main/java/org/smartregister/anc/library/util/Utils.java [hunk 3]
```diff
@@ -225,6 +224,10 @@ public static void proceedToContact(String baseEntityId, HashMap<String, String>
             ContactModel baseContactModel = new ContactModel();
             JSONObject form = baseContactModel.getFormAsJson(quickCheck.getFormName(), baseEntityId, locationId);
 
+            JSONObject globals = new JSONObject();
+            globals.put(ConstantsUtils.CONTACT_NO, personObjectClient.get(DBConstantsUtils.KeyUtils.NEXT_CONTACT));
+            form.put(ConstantsUtils.GLOBAL, globals);
+
```
Comment (dubdabasoduba) [inline opensrp-anc/src/main/java/org/smartregister/anc/library/util/Utils.java:230]: This is a stop gap we need to process quick check globals the same way the other forms globals are processed.  @bennsimon please create an issue of the remaining work and schedule it in

File: opensrp-anc/src/main/java/org/smartregister/anc/library/util/Utils.java [hunk 4]
```diff
@@ -356,7 +359,7 @@ private static String cleanValueResult(String result) {
         List<String> nonEmptyItems = new ArrayList<>();
 
         for (String item : result.split(",")) {
```
Comment (dubdabasoduba) [inline opensrp-anc/src/main/java/org/smartregister/anc/library/util/Utils.java:361]: an NPE might occur if `result` is null.

File: opensrp-anc/src/main/java/org/smartregister/anc/library/activity/ContactJsonFormActivity.java (hunk 2 above)
Comment (bennsimon) [reply opensrp-anc/src/main/java/org/smartregister/anc/library/activity/ContactJsonFormActivity.java:152]: yeah

Comment (bennsimon) [reply opensrp-anc/src/main/java/org/smartregister/anc/library/activity/ContactJsonFormActivity.java:152]: yeah

File: opensrp-anc/src/main/java/org/smartregister/anc/library/util/Utils.java (hunk 3 above)
Comment (bennsimon) [reply opensrp-anc/src/main/java/org/smartregister/anc/library/util/Utils.java:230]: https://github.com/OpenSRP/opensrp-client-anc/issues/512

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/task/LoadContactSummaryDataTask.java  3
- opensrp-anc/src/main/java/org/smartregister/anc/library/repository/PatientRepository.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/util/AncMetadata.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/activity/ContactSummarySendActivity.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5311257&bid=17380449)

**Comment by dubdabasoduba** [comment]
Merged in another branch

**Pull request closed without merging**